pkg/config/reader.go

```diff
@@ -21,19 +22,25 @@ type JobConfigReader struct {
 }
 
 // GetJobConfig retrieves job/config.yaml resource from keptn and parses it into a Config struct.
+// Additionally, also the SHA1 hash of the retrieved configuration will be returned.
 // In case of error retrieving the resource or parsing the yaml it will return (nil,
 // error) with the original error correctly wrapped in the local one
-func (jcr *JobConfigReader) GetJobConfig() (*Config, error) {
+func (jcr *JobConfigReader) GetJobConfig() (*Config, string, error) {
 	resource, err := jcr.Keptn.GetKeptnResource(jobConfigResourceName)
 	if err != nil {
-		return nil, fmt.Errorf("error retrieving job config: %w", err)
+		return nil, "", fmt.Errorf("error retrieving job config: %w", err)
 	}
 
+	hasher := sha1.New()
```

Comment:
could/should we use `sha256` instead of the (AFAIK outdated and insecure) `sha1`?

Reply:
I have the feeling that sha256 is too long to be used as a label value. In this case we usually don't care about security, but just that the configuration file didn't change. But I can check if sha256 or sha224 works.